dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -281,33 +286,41 @@ void CDefaultContextMenu::AddStaticEntry(const HKEY hkeyClass, const WCHAR *szVe
             wcscpy(pEntry->szVerb, szVerb);
         pEntry->hkClass = hkeyClass;
     }
-
-    if (!wcsicmp(szVerb, L"open"))
+	
+    lres = RegOpenKeyExW(hkeyClass, L"shell", 0, KEY_READ, &hShellKey);
+	
+    if(lres == STATUS_SUCCESS)
+        lres = RegQueryValueEx(hShellKey, NULL, 0, &dwType, (LPBYTE)wszDefault, &dwSize);
+	
+    if((!wcsicmp(szVerb, wszDefault) && lres == STATUS_SUCCESS) ||
+       (!wcsicmp(szVerb, L"open") && lres != STATUS_SUCCESS))
     {
-        /* open verb is always inserted in front */
+        /* open verb is always inserted in front, unless the default value of shell entry is set */
         pEntry->pNext = m_pStaticEntries;
         m_pStaticEntries = pEntry;
     }
     else if (pLastEntry)
         pLastEntry->pNext = pEntry;
     else
         m_pStaticEntries = pEntry;
+	
+    RegCloseKey(hShellKey);
 }
 
 void CDefaultContextMenu::AddStaticEntriesForKey(HKEY hKey)
 {
     WCHAR wszName[40];
     DWORD cchName, dwIndex = 0;
     HKEY hShellKey;
-
     LRESULT lres = RegOpenKeyExW(hKey, L"shell", 0, KEY_READ, &hShellKey);
+	
     if (lres != STATUS_SUCCESS)
         return;
 
     while(TRUE)
     {
         cchName = _countof(wszName);
-        if (RegEnumKeyExW(hShellKey, dwIndex++, wszName, &cchName, NULL, NULL, NULL, NULL) != ERROR_SUCCESS)
+        if (RegEnumKeyExW(hShellKey, dwIndex++, wszName, &cchName, NULL, NULL, NULL, NULL) != STATUS_SUCCESS)
```

Comment:
I think we had a little miscommunication here ;)
STATUS_SUCCESS is the incorrect one here (it's for `NTSTATUS`), `ERROR_SUCCESS` is correct (for Win32 errors)

Reply:
Whoops, sorry. I misread that. I'll fix that in a sec.